activation/nipost_test.go

```diff
@@ -114,33 +116,97 @@ func TestNIPostBuilderWithClients(t *testing.T) {
 	poetDb := NewMockpoetDbAPI(gomock.NewController(t))
 	poetDb.EXPECT().GetProof(gomock.Any()).Return(
 		&types.PoetProof{}, &challengeHash, nil,
-	)
-	poetDb.EXPECT().ValidateAndStore(gomock.Any(), gomock.Any()).Return(nil)
-
-	postProvider := newTestPostManager(t)
+	).AnyTimes()
+	poetDb.EXPECT().ValidateAndStore(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
 
-	postCfg := DefaultPostConfig()
-	logger := logtest.New(t).WithName("validator")
-	nipost := buildNIPost(t, postProvider, postCfg, challenge, poetDb)
-	verifier, err := NewPostVerifier(postCfg, logger)
-	r.NoError(err)
-	defer verifier.Close()
-	v := NewValidator(
-		poetDb,
-		postCfg,
-		logger,
-		verifier,
-	)
-	_, err = v.NIPost(
-		context.Background(),
-		postProvider.id,
-		postProvider.commitmentAtxId,
-		nipost,
-		challengeHash,
-		postProvider.opts.NumUnits,
-		verifying.WithLabelScryptParams(postProvider.opts.Scrypt),
-	)
-	r.NoError(err)
+	t.Run("POST with pow creator ID", func(t *testing.T) {
+		t.Parallel()
+		postCfg := DefaultPostConfig()
+		postCfg.PowDifficulty[0] = 64
+		postCfg.MinerIDInK2PowSinceEpoch = 2
+		postProvider := newTestPostManager(t, withPostConfig(postCfg))
+		logger := logtest.New(t).WithName("validator")
+		verifier, err := NewPostVerifier(postProvider.Config(), logger)
+		r.NoError(err)
+		defer verifier.Close()
+
+		nipost := buildNIPost(t, postProvider, postProvider.Config(), challenge, poetDb)
+		v := NewValidator(
+			poetDb,
+			postProvider.Config(),
+			logger,
+			verifier,
+		)
+		_, err = v.NIPost(
+			context.Background(),
+			challenge.PublishEpoch,
+			postProvider.id,
+			postProvider.commitmentAtxId,
+			nipost,
+			challengeHash,
+			postProvider.opts.NumUnits,
+			verifying.WithLabelScryptParams(postProvider.opts.Scrypt),
+		)
+		r.NoError(err)
```

Comment:
this `r.NoError` will fail the parent test, not the subtest. I'd recommend just using `require.NoError(t, err)` instead 🙂